boards : nucleof412zg - CAN Internal controller

pbonazzi opened the pull request:
Signed-off-by: Pietro Bonazzi <pietrobonazzi.open@gmail.com>

gmarull commented on boards/arm/nucleo_f412zg/nucleo_f412zg_defconfig:
```diff
@@ -11,15 +11,19 @@ CONFIG_HW_STACK_PROTECTION=y
-# console
+# Console
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 
-# enable GPIO
+# Enable GPIO
 CONFIG_GPIO=y
 
 # Enable Clocks
 CONFIG_CLOCK_CONTROL=y
 
-# enable pin controller
+# Enable pin controller
 CONFIG_PINCTRL=y
+
+# Enable CAN
+CONFIG_CAN=y
```
this is mixing style changes which are not related. CAN is not a minimal peripheral, it should not be in board defconfig.

pbonazzi replied:
Thank you for your review . I have implemented the changes you requested.

gmarull reviewed (changes requested):
Thanks for your contribution, please, split into at least 2 commits: one that defines new can device, 2nd for board modifications.

FYI https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf

pbonazzi commented:
> Thanks for your contribution, please, split into at least 2 commits: one that defines new can device, 2nd for board modifications.
> 
> FYI https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf

Thank you for your review . I have implemented the changes you requested.

erwango commented:
Can you fix the issues reported here:
https://github.com/zephyrproject-rtos/zephyr/runs/6916504029?check_suite_focus=true

Also, check this https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf, starting slide 21

FRASTM commented on dts/arm/st/f4/stm32f412.dtsi:
```diff
@@ -190,5 +190,18 @@
+			interrupt-names = "TX", "RX0", "RX1", "SCE";
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x02000000>;
+			status = "disabled";
+			label = "CAN_1";
```
'label' is no more relevant

FRASTM commented on dts/arm/st/f4/stm32f412.dtsi:
```diff
@@ -190,5 +190,18 @@
 			status = "disabled";
 			label = "QUADSPI";
 		};
+
```
extra line to remove

FRASTM reviewed (commented):
Also take this opportunity to add the` can2: can@40006800`

pbonazzi commented:
> Also take this opportunity to add the` can2: can@40006800`

Continue conversation in #49411
